bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/handler/SoulissT42Handler.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.souliss.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.souliss.SoulissBindingConstants;
+import org.openhab.binding.souliss.SoulissBindingProtocolConstants;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.PrimitiveType;
+
+/**
+ * The {@link SoulissT42Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Tonino Fazio - Initial contribution
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+
+@NonNullByDefault
+public class SoulissT42Handler extends SoulissGenericHandler {
+    @Nullable
+    Configuration gwConfigurationMap;
+    byte t4nRawState;
+
+    public SoulissT42Handler(Thing thing) {
+        super(thing);
+    }
+
+    // called on every status change or change request
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getAsString().split(":")[3].equals(SoulissBindingConstants.T4N_REARMALARM_CHANNEL)) {
+            if (command instanceof OnOffType) {
+                switch (command.toFullString()) {
+                    case "ON":
```

Comment:
switch (command) {
	                        case OnOffType.ON:

More efficient to compare the object and not turn it into a string and then compare.

Reply:
your edit not working on switch case statement , error on command type (only int or primitives)

Reply:
See two review comments above for the fix in the syntax.

Reply:
fixed